Source/WebCore/rendering/style/RenderStyle.cpp

```diff
@@ -2108,6 +2110,31 @@ int RenderStyle::computeLineHeight(const Length& lineHeightLength) const
     return clampTo<int>(lineHeightLength.value());
 }
 
+// FIXME: Remove this after all old calls to whiteSpace() are replaced with appropriate
+// calls to whiteSpaceCollapse() and textWrap().
+WhiteSpace RenderStyle::whiteSpace() const
+{
+    auto whiteSpaceCollapse = static_cast<WhiteSpaceCollapse>(m_inheritedFlags.whiteSpaceCollapse);
+    auto textWrap = static_cast<TextWrap>(m_inheritedFlags.textWrap);
+    if (whiteSpaceCollapse == WhiteSpaceCollapse::BreakSpaces && textWrap == TextWrap::Wrap)
+        return WhiteSpace::BreakSpaces;
+    if (whiteSpaceCollapse == WhiteSpaceCollapse::Collapse && textWrap == TextWrap::Wrap)
+        return WhiteSpace::Normal;
+    if (whiteSpaceCollapse == WhiteSpaceCollapse::Collapse && textWrap == TextWrap::NoWrap)
+        return WhiteSpace::NoWrap;
+    if (whiteSpaceCollapse == WhiteSpaceCollapse::Preserve && textWrap == TextWrap::NoWrap)
+        return WhiteSpace::Pre;
+    if (whiteSpaceCollapse == WhiteSpaceCollapse::PreserveBreaks && textWrap == TextWrap::Wrap)
+        return WhiteSpace::PreLine;
+    if (whiteSpaceCollapse == WhiteSpaceCollapse::Preserve && textWrap == TextWrap::Wrap)
+        return WhiteSpace::PreWrap;
+
+    // Reachable only if white-space-collapse/text-wrap properties were individually set
+    // instead of being changed via white-space shorthand keyword. Do nothing for now
+    // since this is a temporary function.
```

Comment:
// Reachable for combinations that can't be represented with the white-space syntax.

would be more accurate.